st2client/tests/unit/test_commands.py

```diff
@@ -246,34 +246,51 @@ def test_command_delete_failed(self):
         self.assertRaises(Exception, self.branch.commands['delete'].run, args)
 
 
-class ActionExecutionReadCommandTestCase(unittest2.TestCase):
+class ResourceViewCommandTestCase(unittest2.TestCase):
 
-    def test_get_exclude_attributes(self):
+    def setUp(self):
+        ResourceViewCommand.display_attributes = []
+
+    def test_get_include_attributes(self):
         cls = namedtuple('Args', 'attr')
 
         args = cls(attr=[])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, ['result', 'trigger_instance'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, [])
 
         args = cls(attr=['result'])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, ['trigger_instance'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, ['result'])
 
         args = cls(attr=['result', 'trigger_instance'])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, [])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, ['result', 'trigger_instance'])
 
         args = cls(attr=['result.stdout'])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, ['trigger_instance'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, ['result.stdout'])
 
         args = cls(attr=['result.stdout', 'result.stderr'])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, ['trigger_instance'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, ['result.stdout', 'result.stderr'])
 
         args = cls(attr=['result.stdout', 'trigger_instance.id'])
-        result = ActionExecutionReadCommand._get_exclude_attributes(args=args)
-        self.assertEqual(result, [])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, ['result.stdout', 'trigger_instance.id'])
+
+        ResourceViewCommand.display_attributes = ['id', 'status']
+
+        args = cls(attr=[])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(set(result), set(['id', 'status']))
+
+        args = cls(attr=['trigger_instance'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(set(result), set(['trigger_instance']))
+
+        args = cls(attr=['all'])
+        result = ResourceViewCommand._get_include_attributes(args=args)
+        self.assertEqual(result, None)
```

Comment:
Can  you add a test case for a field that doesn't exist in the dict? Also a test case for some existing and some not existing in the same call?

Reply:
That would be an integration / end to end test.

Will add it to st2tests.

But yeah, in case user specifies an nonexistent field, API will return 400 error and that will be propagated to the CLI (we already have tests for that for st2api).

Reply:
@lakshmi-kannan Added end to end tests in - https://github.com/StackStorm/st2tests/pull/146.